### PR TITLE
50707 replacpt sample

### PR DESCRIPTION
**Pull request by mikerhodes**
_What_

Update the Replication Acceptance and Sample projects to use the new CDTDocumentRevision classes.

_Why_

So we can still run the sample and acceptance tests.

_How_

* d5a3fe1cb3c85519ed022cc8232738629d31798b adds real model classes to the Todo project, wrapping the document interaction. These classes handle modifying documents as required, and providing revisions to be saved.
* 56d9847b17345f12bd6766260714a292c0166c34 basically rewrites any `mutableCopy` calls to no-ops. It also calls `[CDTDocumentRevision revisionWithDocId:blah]` instead of `[CDTMutableDocumentRevision revision]; revision.docId = blah;` as `docId` is a readonly property of CDTDocumentRevision, and that can't be changed.

reviewer @rhyshort 
reviewer @tomblench

### CODE REVIEW

File: Project/Project/CDTTodo.h [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 #import <Foundation/Foundation.h>
 
+@class CDTDocumentRevision;
+
 /**
  A simple class that maps from a JSON document to a Todo task, to
```
Comment (tomblench) [inline Project/Project/CDTTodo.h]: Update comment to reflect that this now wraps a revision?

File: Project/Project/CDTViewController.m [hunk 2]
```diff
@@ -25,7 +25,7 @@ @interface CDTViewController ()
 
 @property (readonly) CDTDatastore *datastore;
 @property (readonly) CDTTodoReplicator *todoReplicator;
-@property (nonatomic,strong) NSArray *taskRevisions;
+@property (nonatomic, strong) NSArray *taskRevisions;
```
Comment (tomblench) [inline Project/Project/CDTViewController.m]: This variable should be renamed to reflect the fact it is now an array of `CDTTodo`s

Comment (mikerhodes) [inline Project/Project/CDTViewController.m]: Just `tasks` or `todos` do you think?

Comment (tomblench) [inline Project/Project/CDTViewController.m]: `todos` or `todoList` maybe?

Comment (mikerhodes) [inline Project/Project/CDTViewController.m]: Done in f602efd

File: Project/Project/CDTTodo.h (hunk 1 above)
Comment (mikerhodes) [inline Project/Project/CDTTodo.h]: Updated in 481ea75

**Comment by tomblench** [comment]
OK +1

**Comment by rhyshort** [comment]
+1

**Pull request merged**